src/service/viewport-impl.js

```diff
@@ -453,6 +453,22 @@ export class Viewport {
   }
 
   /*
+   * Instruct the viewport to enter sidebar mode.
+   */
+  enterSidebarMode() {
```

Comment:
I wouldn't add this to viewport. It is fine for sidebar component to call these two methods directly